### PR TITLE
fix: Rename the Prometheus Input Plugin Timeout variable

### LINKED ISSUE #12565: The timeout parameter inside the Prometheus plugin is wrongly named to parse the timeout directly from the configuration
### System info

Telegraf 1.25.0

### Change request

It only changes inside a code line to make it not necessary to forward the timeout and to use it directly from the configuration. 

### Expected behavior

The parameter should be parsed directly and not converted like [this](https://github.com/influxdata/telegraf/blob/4b1b74fefff050e034cb69656e730c11db150c6e/plugins/inputs/prometheus/prometheus.go#L166) line handle it.

### Actual behavior

I've prepared a fix for the [root cause](https://github.com/influxdata/telegraf/issues/12485) to solve the corresponding case. The parameter is forwarded to the corresponding library to not break someone's use case. In future it would be better to parse the configuration as timeout directly from the forwarded configuration.

### Background

We've already discussed the topic here: https://github.com/influxdata/telegraf/pull/12559#discussion_r1089055806

From my understanding, the parameter should be replaced in the future with a new name called `timeout` to directly parse the configuration. I've decided to handle it not directly inside this [PR](https://github.com/influxdata/telegraf/pull/12559#pullrequestreview-1275196432) because it's a breaking change.

### PR DESCRIPTION
**Pull request by ZPascal**
Rename the Timeout Prometheus Input Plugin variable inside the configuration and the code to parse the corresponding value directly

**Be aware it's a breaking change for older versions and should be merged into the next major version.**

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12565

Fix the corresponding case and renames the variable as already discussed inside #12565

### CODE REVIEW

**Review by powersj — COMMENTED** [review]
Thanks for taking the time to quickly put a PR.

I am happy to see the new option, but instead of outright removing the old option, we want to [deprecate the option](https://github.com/influxdata/telegraf/blob/master/docs/developers/DEPRECATION.md#deprecate-options). This way users get a warning before we outright remove.

**Comment by ZPascal** [comment]
@powersj You're right. That makes sense. I'll handle it later.

File: plugins/inputs/prometheus/README.md [hunk 1]
```diff
@@ -126,8 +126,12 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
   # headers = {"X-Special-Header" = "Special-Value"}
 
   ## Specify timeout duration for slower prometheus clients (default is 3s)
+  ##   deprecated in 1.26; use the timeout option
   # response_timeout = "3s"
 
+  ## Specify timeout duration for slower prometheus clients (default is 3s)
+  # timeout = "3s"
+
```
Comment (powersj) [inline plugins/inputs/prometheus/README.md:134]: Let's flip the order so the new option shows up first:

```suggestion
  ## Specify timeout duration for slower prometheus clients (default is 3s)
  # timeout = "3s"
  
  ##   deprecated in 1.26; use the timeout option
  # response_timeout = "3s"
  
```

File: plugins/inputs/prometheus/sample.conf [hunk 2]
```diff
@@ -109,8 +109,12 @@
   # headers = {"X-Special-Header" = "Special-Value"}
 
   ## Specify timeout duration for slower prometheus clients (default is 3s)
+  ##   deprecated in 1.26; use the timeout option
   # response_timeout = "3s"
 
+  ## Specify timeout duration for slower prometheus clients (default is 3s)
+  # timeout = "3s"
+
```
Comment (powersj) [inline plugins/inputs/prometheus/sample.conf:117]:
```suggestion
  ## Specify timeout duration for slower prometheus clients (default is 3s)
  # timeout = "3s"
  
  ##   deprecated in 1.26; use the timeout option
  # response_timeout = "3s"
  
```

**Review by powersj — COMMENTED** [review]
Thanks for the update and tests! Just two suggestions to make the new option show up first in the config.

**Comment by ZPascal** [comment]
@powersj Sure, it's done.

**Review by powersj — APPROVED** [review]
Thanks again!

File: plugins/inputs/prometheus/sample.conf [hunk 3]
```diff
@@ -109,8 +109,11 @@
   # headers = {"X-Special-Header" = "Special-Value"}
 
   ## Specify timeout duration for slower prometheus clients (default is 3s)
+  # timeout = "3s"
+  
+  ##   deprecated in 1.26; use the timeout option
   # response_timeout = "3s"
-
+
```
Comment (srebhan) [inline plugins/inputs/prometheus/sample.conf:116]: I would say remove the option in the `sample.conf` to have the example show the recommended way of configuration but _keep_ the option in code. @powersj  what do you think?

File: plugins/inputs/prometheus/prometheus_test.go [hunk 4]
```diff
@@ -196,6 +196,63 @@ func TestPrometheusGeneratesMetricsSlowEndpointHitTheTimeout(t *testing.T) {
...
+	require.NoError(t, err)
+
+	var acc testutil.Accumulator
+
+	err = acc.GatherError(p.Gather)
+	errMessage := fmt.Sprintf("error making HTTP request to %s/metrics: Get \"%s/metrics\": "+
+		"context deadline exceeded (Client.Timeout exceeded while awaiting headers)", ts.URL, ts.URL)
+	errExpected := errors.New(errMessage)
+	require.Equal(t, errExpected, err)
+	require.Error(t, err)
```
Comment (srebhan) [inline plugins/inputs/prometheus/prometheus_test.go:253]: How about
```suggestion
	require.ErrorContains(t, err, "error making HTTP request to "+ts.URL+"/metrics")
```

**Review by srebhan — COMMENTED** [review]
Thanks for the fix @ZPascal. I do have two small comments...

File: plugins/inputs/prometheus/prometheus_test.go (hunk 4 above)
Comment (srebhan) [reply plugins/inputs/prometheus/prometheus_test.go:253]: @ZPascal any chance you can fix this last comment so we can merge this PR?

Comment (ZPascal) [reply plugins/inputs/prometheus/prometheus_test.go]: Sure. Thank you for the optimization.

**Review by srebhan — APPROVED** [review]
Thanks for your contribution @ZPascal!

**Pull request merged**